api/v1alpha1/promotion_types.go

```diff
@@ -207,6 +207,9 @@ type PromotionStep struct {
 	Task *PromotionTaskReference `json:"task,omitempty" protobuf:"bytes,5,opt,name=task"`
 	// As is the alias this step can be referred to as.
 	As string `json:"as,omitempty" protobuf:"bytes,2,opt,name=as"`
+	// If is an optional expression that, if present, must evaluate to a boolean
+	// value. If the expression evaluates to false, the step will be skipped.
+	If string `json:"if,omitempty" protobuf:"bytes,7,opt,name=if"`
```

Comment:
Would it be worth describing what happens if it doesn't evaluate to a boolean? As a reader who hasn't looked at the code, I might wonder if it's an error or is implicitly false.